### PR TITLE
Add stable release maintainers to CODEOWNERS

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Update 3.0 branch CODEOWNERS to require CBL-Mariner-Stable-Maintainers
team review for all files in this branch. This will include the 3.0-dev
branch and eventually the 3.0 release and fast-track branches.

SPECS-EXTENDED will still require cbl-mariner-devs to review due to the
active changes going on in that directory.

Signed-off-by: Chris Co <chrco@microsoft.com>

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,98 +1,2 @@
...
-
-# Image Customizer
-/toolkit/tools/imagecustomizer/ @microsoft/cbl-mariner-imagecustomizer
-/toolkit/tools/imagecustomizerapi/ @microsoft/cbl-mariner-imagecustomizer
-/toolkit/tools/osmodifier @microsoft/cbl-mariner-imagecustomizer
-/toolkit/tools/pkg/imagecustomizerlib/ @microsoft/cbl-mariner-imagecustomizer
-/toolkit/tools/pkg/osmodifierlib @microsoft/cbl-mariner-imagecustomizer
-/toolkit/scripts/build_tag_imagecustomizer.mk @microsoft/cbl-mariner-imagecustomizer
+# For stable release branches, ensure stable release maintainers are added as code reviewers
+* @microsoft/cbl-mariner-stable-maintainers
```
Comment (PawelWMS) [inline .github/CODEOWNERS:2]: Why do we want that in 3.0? And why remove code owners of other areas? We're going to miss all the notifications for the code and packages we own, while at the same time the maintainers will most probably get overwhelmed with PRs to review. We already have severe review delays in 2.0, so I'm a bit worried about this change.

Comment (mfrw) [inline .github/CODEOWNERS:2]: Can we make the change only for SPECS.
There is an ongoing effort to upgrade SPECS-EXTENDED.

Comment (christopherco) [reply .github/CODEOWNERS:2]: Discussed offline - summary is we're going to start with confining the stable-maintainer reviewers to the SPECS folder first, and select portions of the toolkit (specifically docs, manifests, imageconfig definitions)

Comment (christopherco) [reply .github/CODEOWNERS:2]: ACK - will start with SPECS first, and will include SPECS-EXTENDED in a future PR

**Review by eric-desrochers — APPROVED** [review]
LGTM as per discussion in stable shiproom

**Pull request merged**